recipes/tmap/meta.yaml

```diff
@@ -0,0 +1,52 @@
+{% set name = "tmap" %}
+{% set version = "0.1.0" %}
+
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/reymond-group/tmap/archive/9aa043dd1af022161048ee0a4b5554ec5e34977f.zip
```

Comment:
I appears that this recipe is pulling a source archive from a git commit and labeling that version as 0.1.0? Unless I have misunderstood, we cannot merge `tmap` in this form.

Reply:
I'm not sure were the "true source" of versioning for that package might be.
https://anaconda.org/tmap/tmap/files might give some indications.

Reply:
Indeed there is not git version in the source repo. I opened a ticket for it at https://github.com/reymond-group/tmap/issues/14 but the repo does not seem to be maintained. I don't really know what to do about that. Should I just fork it, tag something, and use this fork?

Reply:
Good find @mbargull! I looked at a copy of the recipe in the conda packages. They were building straight from master. :(

Reply:
What about a 0.0.0 version? Have you seen such cases in the past?

Reply:
As long as the commit is fixed in the recipe, there is a way to come back to it. Of course, a proper git version would be best.

Reply:
You could download https://anaconda.org/tmap/tmap/1.0.4/download/linux-64/tmap-1.0.4-py37h6bb024c_0.tar.bz2 and try to deduct to which Git commit that release corresponds to.
... I know that it's stupid work that no one should have to do. But I don't see how upstream gives us another choice here.
Frustating

Reply:
Good idea. I am fine having a look at this.

Reply:
I am ok with this @margbull but I do think we should be shipping some sort of dev version, say `1.0.5.dev0` and then incrementing only the dev number (e.g., `1.0.5.dev1`, `1.0.5.dev2`) etc. 

I am personally more happy with us shipping `0.0.0.dev0`, `0.0.0.dev1` etc in this case since we don't know the intent of what is on the tmap channel or of the upstream authors.

Reply:
Note that crosstool-ng embeds the git commit right in the version, though this is much more specialized due to it building a compiler: https://github.com/conda-forge/crosstool-ng-feedstock/blob/master/recipe/meta.yaml#L1

Reply:
@mbargull if you're fine with that I also prefer using `0.0.1.dev0`. `1.0.4` is only present int he tmap conda channel. The `setup.py` has the version set to `1.0.0`... Also I couldn't find any git hash in https://anaconda.org/tmap/tmap/1.0.4/download/linux-64/tmap-1.0.4-py37h6bb024c_0.tar.bz2 (but I could still look at the uploaded date of the package).

Reply:
> Note that crosstool-ng embeds the git commit right in the version, though this is much more specialized due to it building a compiler: https://github.com/conda-forge/crosstool-ng-feedstock/blob/master/recipe/meta.yaml#L1

I thought about doing that but since the git version is already present in the recipe and shipped with the package using `dev0` seems ok to me. Let me know if you prefer using a git hash.

Reply:
If we were able to make out the source commit, then I'd go for the version they used on their channel.
But yeah, it seems this package is for the most part just compiled code, which makes the above difficult.
So, yes, I agree, just go for the "we don't know what the heck upstream does"-version number, `0.0.1.dev0`.

Reply:
Go for `0.0.1.dev0` then. xref https://github.com/conda-forge/staged-recipes/pull/12498/commits/5ce0a55d15d4288d87255d86b30e082102454ec9